src/messages.rs

```diff
@@ -194,6 +194,11 @@ pub struct GroupList {
 }
 
 impl GroupList {
+    /// Gets the `route`-th name in the group list.
+    pub fn get_routeth_name(&self, dst_name: &XorName, route: usize) -> &XorName {
+        RoutingTable::get_routeth_name(self.pub_ids.iter().map(|id| id.name()), dst_name, route)
+    }
```

Comment:
Now that we're using `GroupList::get_routeth_name` in _both_ cases, it seems we don't need the method in the routing table anymore at all?
Also, instead of converting everything to names, this chould just return the `route`th `PublicId` instead. We can then call `name()` on it in `get_signature_target`.